### PR TITLE
EES-1447 - Create release time identifier error

### PR DESCRIPTION
**Pull request by bennettstuart**
[Ticket 1447](https://dfedigital.atlassian.net/browse/EES-1447)
Fixes:
- Added additional validation to year field, checking year has 4 characters

image

### CODE REVIEW

File: src/explore-education-statistics-admin/src/pages/release/components/ReleaseSummaryForm.tsx [hunk 1]
```diff
@@ -102,7 +102,13 @@ const ReleaseSummaryForm = <
 
   const baseSchema = Yup.object<ReleaseSummaryFormValues>({
     timePeriodCoverageCode: Yup.string().required('Choose a time period'),
-    timePeriodCoverageStartYear: Yup.string().required('Enter a year'),
+    timePeriodCoverageStartYear: Yup.string()
+      .required('Enter a year')
+      .test(
+        'length',
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/pages/release/components/ReleaseSummaryForm.tsx:108]: No need for a custom validator, there's literally a `length` method built into `Yup.string`.

File: src/explore-education-statistics-admin/src/pages/release/components/ReleaseSummaryForm.tsx [hunk 2]
```diff
@@ -102,7 +102,13 @@ const ReleaseSummaryForm = <
 
   const baseSchema = Yup.object<ReleaseSummaryFormValues>({
     timePeriodCoverageCode: Yup.string().required('Choose a time period'),
-    timePeriodCoverageStartYear: Yup.string().required('Enter a year'),
+    timePeriodCoverageStartYear: Yup.string()
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/pages/release/components/ReleaseSummaryForm.tsx:105]: There's currently no unit tests for this form component, so let's write some.

**Review by ntsim — CHANGES_REQUESTED** [review]

File: src/explore-education-statistics-admin/src/pages/release/components/ReleaseSummaryForm.tsx (hunk 2 above)
Comment (bennettstuart) [reply src/explore-education-statistics-admin/src/pages/release/components/ReleaseSummaryForm.tsx:105]: added tests:
  ReleaseSummaryForm
    √ renders correctly (104ms)
    √ validation error render where appropriate (278ms)
    √ renders with provided initial values (31ms)
    √ submits form with valid values (133ms)

**Review by ntsim — CHANGES_REQUESTED** [review]

File: src/explore-education-statistics-admin/src/pages/release/components/__tests__/ReleaseSummaryForm.test.tsx [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+    );
+
+    await waitFor(() => {
+      expect(
+        screen.getByText('Select time period coverage'),
+      ).toBeInTheDocument();
+    });
+
+    const selectYearType = screen.getByLabelText('Type', {
+      selector: 'select',
```
[the remark this answers is not in the document]
Comment (markhiveit) [reply src/explore-education-statistics-admin/src/pages/release/components/__tests__/ReleaseSummaryForm.test.tsx:67]: I've removed them where I can. There are few links that have the same text as a span.

**Review by ntsim — APPROVED** [review]

**Pull request merged**